pkg/istio-agent/agent.go

```diff
@@ -321,6 +322,15 @@ func (a *Agent) initializeEnvoyAgent(_ context.Context) error {
 		}
 		a.envoyOpts.ConfigPath = out
 		a.envoyOpts.ConfigCleanup = true
+
+		bytes, err := proto.Marshal(node.Metadata.ToStruct())
```

Comment:
I thought we used flatbuffers? does it encode to the same or am I just wrong about that?

Reply:
Flatbuffers is an internal representation, that we can remove without user facing changes.

Reply:
What about cross-version calls?

Reply:
We never flatbuffer on the wire if that's the question.

Reply:
Why still use flat buffers then? Would be nice if we could standardize on proto if perf impact is minimal (not for this PR obviously)

Reply:
It was used because Wasm can't handle protos efficiently. Now that we don't recommend Wasm for telemetry, it could be a cleanup, although some users are probably still using internal flatbuffers and would be impacted - including our own SD extension.

Reply:
If we are going to be breaking the internal protocol between proxies this needs broad discussion around the cost vs benefits.

Reply:
@howardjohn There's no difference on the network between this and before. Flatbuffers were always internal to Istio and never sent on the wire. I don't think I understand the concern here.

Reply:
Oh so the wire format is the same?

We can do 
```
meta := ...
roundtrip := flatbuffer_unmarshal(proto_marshal(meta))
meta == roundtrip
```

?

If so, no concern

Reply:
Yeah, flatbuffer is just a subset of struct metadata with 4-5 fields and converted internally in Envoy from proto to flatbuffers because Wasm. We can construct the subset here as well, and not even use flatbuffers to roundtrip.